message_app/model.py

```diff
@@ -61,3 +69,107 @@ def get_last_user_id(cls) -> int:
         # Get last user ID
         id = User.query.order_by(User.id.desc()).first().id
         return id
+
+# SQLAlchemy model for conversation_request table
+class ConversationRequest(db.Model):
+    __tablename__ = 'conversation_request'
+    id = db.Column(db.Integer, primary_key=True)
+    initiator_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=False)
+    receiver_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=False)
+    accepted = db.Column(db.Integer, nullable=False, default=0)
+    request_time = db.Column(db.Integer, nullable=False)
+    accepted_time = db.Column(db.Integer)
+
+    def __eq__(self, other_request: CoversationRequest) -> bool:
+        # Compare two users using its username
+        return other_request.id == self.id
+
+    # Return a json encoding of the user data
+    def to_json(self) -> str:
+        data = {
+            "id": self.id,
+            "initiator_id": self.initiator_id,
+            "receiver_id": self.receiver_id,
+            "accepted": self.accepted,
+            "request_time": self.request_time,
+            "accepted_time": self.accepted_time
+        }
+        return json.dumps(data)
+
+#    def accept(self, time: int) -> None:
+#        # Accept a conversation request
+#        self.accepted = 1
+#        self.accepted_time = time
+#        db.seesion.commit()
+#
+#    def reject(self) -> None:
+#        # Reject a conversation request
+#        self.accepted = 0
+#        db.session.commit()
```

Comment:
Why are these commented out?

Reply:
I changed them to class methods. At first, I tried to make them instance methods but then I got an error "object has no attribute session" (https://stackoverflow.com/questions/20201809/sqlalchemy-flask-attributeerror-session-object-has-no-attribute-model-chan). I do not quite understand that.

Reply:
Do you still have the instance version? Can you paste it here?

Reply:
those commanded code are instance method version. the class methods version are bellow

Reply:
There is a typo in your accept function. Is this the reason?

Reply:
Another reason that I can think of is that there are no database operations in your instance functions before you call commit. 
For the class functions, they all have a database operation carried out, so there is a `Session` created for those operations. It seems to me that there is no `Session` created and attached to `db`

Reply:
Keep in mind that the StackOverflow link you included here is another problem. Here we do not have the `Session `on the SQLAlchemy instance (`db`)

Reply:
oh yah, it is a type error :)) My bad :))

Reply:
From your latest [commit](https://github.com/leth02/VL_Chat_App/pull/17/commits/b3961bee3da623177965150a10707e2312dc83dc), I see that instance methods are working now? @vuongtranluc

Reply:
yes, it's working now.

Reply:
I also deleted the class methods version